ui/app/src/components/shared/airlock/AirlockNewRequest.tsx

```diff
@@ -0,0 +1,219 @@
+    },
+    [setNewRequest]
+  );
+
+  useEffect(
+    () => setRequestValid(newRequest.businessJustification?.length > 0),
+    [newRequest, setRequestValid]
+  );
+
+  // Submit Airlock request to API
+  const create = useCallback(async () => {
+    if (requestValid) {
+      setCreating(true);
+      setCreateError(false);
+      try {
+        const response = await apiCall(
+          `${ApiEndpoint.Workspaces}/${workspaceCtx.workspace.id}/${ApiEndpoint.AirlockRequests}`,
+          HttpMethod.Post,
+          workspaceCtx.workspaceApplicationIdURI,
+          newRequest
+        );
+        props.onCreateRequest(response.airlockRequest);
+        setHideCreateDialog(true);
+      } catch (err: any) {
+        err.userMessage = 'Error submitting airlock request';
+        setApiSubmitError(err);
+        setCreateError(true);
+      }
+      setCreating(false);
+    }
+  }, [apiCall, newRequest, props, workspaceCtx, requestValid]);
+
+  const dismissPanel = useCallback(() => navigate('../'), [navigate]);
+
+  let title;
+  let currentStep;
+
+  // Render current step depending on whether requestType has been selected
+  if (!newRequest.requestType) {
+    title = "New airlock request";
+    currentStep = <Stack style={{marginTop: '40px'}} tokens={stackTokens}>
+      <DocumentCard
+        aria-label="Import"
+        type={DocumentCardType.compact}
+        onClick={() => setNewRequest({ requestType: AirlockRequestType.Import } as NewAirlockRequest)}>
+        <DocumentCardPreview {...importPreviewGraphic} />
+        <DocumentCardDetails>
+          <DocumentCardTitle title="Import" styles={cardTitleStyles} />
+          <DocumentCardTitle
+            title="Import files into a workspace from outside of the TRE."
+            shouldTruncate
+            showAsSecondaryTitle
+          />
+        </DocumentCardDetails>
+      </DocumentCard>
+
+      <DocumentCard
+        aria-label="Export"
+        type={DocumentCardType.compact}
+        onClick={() => setNewRequest({ requestType: AirlockRequestType.Export } as NewAirlockRequest)}>
+        <DocumentCardPreview {...exportPreviewGraphic} />
+        <DocumentCardDetails>
+          <DocumentCardTitle title="Export" styles={cardTitleStyles} />
+          <DocumentCardTitle
+            title="Export files from a workspace to the outside world."
+            shouldTruncate
+            showAsSecondaryTitle
+          />
+        </DocumentCardDetails>
+      </DocumentCard>
+    </Stack>;
+  } else {
+    title = `New airlock ${newRequest.requestType} request`;
+    currentStep = <Stack style={{marginTop: '40px'}} tokens={stackTokens}>
+      <TextField
+        label="Business Justification"
+        placeholder="Enter a justification for your request."
+        value={newRequest.businessJustification}
+        onChange={onChangeBusinessJustification}
+        multiline
+        rows={10}
+        required
+      />
+    </Stack>;
+  }
+
+  const renderFooter = useCallback(() => {
```

Comment:
think any hooks should be defined at the top of the function, definitely before any conditional logic